src/sentry/testutils/outbox.py

```diff
@@ -89,12 +91,28 @@ def assert_webhook_payloads_for_mailbox(
         assert message.request_body == expected_payload["request_body"]
         assert message.schedule_for == THE_PAST
         assert message.attempts == 0
-        assert message.region_name is not None
-        try:
-            region_names_set.remove(message.region_name)
-        except KeyError:
-            raise Exception(
-                f"Found ControlOutbox for '{message.region_name}', which was not in region_names: {str(region_names_set)}"
-            )
+
+        if destination_types:
+            destination_type = DestinationType(message.destination_type)
+            assert destination_type in destination_types
+            destination_types[destination_type] -= 1
+            if destination_types[destination_type] == 0:
+                del destination_types[destination_type]
+
+        if message.destination_type == DestinationType.CODECOV:
+            assert message.region_name is None
+        else:
+            assert message.region_name is not None
+            try:
+                region_names_set.remove(message.region_name)
+            except KeyError:
+                raise Exception(
+                    f"Found ControlOutbox for '{message.region_name}', which was not in region_names: {str(region_names_set)}"
+                )
     if len(region_names_set) != 0:
         raise Exception(f"WebhookPayload not found for some region_names: {str(region_names_set)}")
+
+    if destination_types and len(destination_types) != 0:
+        raise Exception(
+            f"WebhookPayload not found for some destination_types: {str(destination_types)}"
+        )
```

Comment:
I think this error can be improved, It's actually that not enought webhookPayloads were found for a given destination type right, not that they weren't found at all
```suggestion
        exc_strs = [f"Missing {destination_count} WebhookPayloads for {destination_type} destination. \n" for [destination_type, destination_count] in destination_types]
        raise Exception(''.join(exc_strs))
```

best i could do in github's markdown editor lol, feel free to do your own thing